raster-foundry/raster-foundry
# Include platform id in access logs

## Overview
Include platform id in access logs

### Checklist


### Notes
This _should_ be working in its current state with separate logs for platform id per tile request:
```
backsplash_1              | [raster-io-36] INFO  c.r.b.s.Authenticators - platformId = 31277626-968b-4e40-840b-559d9c67863c
backsplash_1              | [raster-io-39] INFO  c.r.b.s.Authenticators - platformId = 31277626-968b-4e40-840b-559d9c67863c
backsplash_1              | [raster-io-36] INFO  i.j.Configuration - Initialized tracer=JaegerTracer(version=Java-1.0.0, serviceName=raster-foundry, reporter=CompositeReporter(reporters=[RemoteReporter(sender=UdpSender(), closeEnqueueTimeout=1000), LoggingReporter(logger=Logger[io.jaegertracing.internal.reporters.LoggingReporter])]), sampler=ConstSampler(decision=true, tags={sampler.type=const, sampler.param=true}), tags={hostname=17fde6bda7e3, jaeger.version=Java-1.0.0, ip=172.25.0.7}, zipkinSharedRpcSpan=false, expandExceptionLogs=false, useTraceId128Bit=false)
backsplash_1              | [raster-io-39] INFO  i.j.Configuration - Initialized tracer=JaegerTracer(version=Java-1.0.0, serviceName=raster-foundry, reporter=CompositeReporter(reporters=[RemoteReporter(sender=UdpSender(), closeEnqueueTimeout=1000), LoggingReporter(logger=Logger[io.jaegertracing.internal.reporters.LoggingReporter])]), sampler=ConstSampler(decision=true, tags={sampler.type=const, sampler.param=true}), tags={hostname=17fde6bda7e3, jaeger.version=Java-1.0.0, ip=172.25.0.7}, zipkinSharedRpcSpan=false, expandExceptionLogs=false, useTraceId128Bit=false)
```

This is a very big diff, but most of the changes are mechanically updating code that previously consumed a `User` instance to now accept a `(User, Option[UUID])` so that we can eventually log the platform id since we really only use it in one place.

The real meaningful changes to look at are in `UserDao`, `Main` (backsplash), and `Authenticators`. The actual logging of the platform id happens at the bottom of `Authenticators.scala` since everything's been changed to work with a `UserMaybePlatformId` but all of the downstream code will just want a `User`. It's at that point where we log the platform id as a side-effect and then only pass the user along. I'm not totally sure if that's the exact right place to be doing that, but it does seem to work and it does it for all of the tile server endpoints we care about. That said, I'm not totally sure what precisely constitutes a tile request in terms of use charging the client for it. We may want to do this a little differently -- another middleware for platform id logging may be preferable, or perhaps trying to incorporate this into the existing access log middleware -- but I think the general approach of "retrieve and cache a platform id if available along with the authenticated user and account for that everywhere" should serve our purposes regardless of whether the platform logging middleware implementation changes substantially. It's possible I got this wrong and we should definitely check we're logging the platform id based on the correct access event.

In one of the later commits I switched over to only caching the platform id instead of the whole platform due to a decoding runtime error I ran into:
```
api-server_1              | Caused by: io.circe.DecodingFailure$$anon$2: Attempt to decode value on failed cursor: DownField(privateSettings),DownN(1)
```
I did this mostly for expediency's sake given the fact that we likely only care about the platform id and nothing else about the platform. However, I think fixing the decoder could allow us to store the whole `Platform` instead of just the id if we're so inclined.

Lastly, I think this change will require busting the memcached cache.

## Testing Instructions

- How to test this PR
- Prefer bulleted description
- Start after checking out this branch
- Include any setup required, such as bundling scripts, restarting services, etc.
- Include test case, and expected output

Closes [#1370](https://app.zenhub.com/workspaces/raster-foundry-team-workspace-5878d68005907de543b50bbd/issues/azavea/raster-foundry-platform/1370)
